oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/IntMatch.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.core.analysis.metrics.expression;
+
+import org.apache.skywalking.oap.server.core.analysis.metrics.annotation.FilterMatcher;
+
+@FilterMatcher
+public class IntMatch {
+
+    public boolean match(int left, int right) {
+        return left == right;
+    }
+
+    public boolean match(Integer left, Integer right) {
```

Comment:
My point is this is not working. Consider to remove this and run test.

Reply:
![image](https://user-images.githubusercontent.com/45945752/131852708-7fd9cce2-a22e-4349-b733-035c5073622a.png)
The problem is the automatic binning, which is fine under my test

Reply:
This could work, but it still requires you to change g4 file, because it was `stringMatch`, which doesn't fit the case.

Reply:
My point is NumberEqual and IntEqual both make sense and clear. You could add `match(long left, long right` in case for the future. `StringMatch` should be kept in String only.